selectors/views/channel_sidebar.ts

```diff
@@ -47,6 +47,18 @@ export const getCategoriesForCurrentTeam: (state: GlobalState) => ChannelCategor
     });
 })();
 
+export const getAutoSortedCategoryIds: (state: GlobalState) => string[] = (() => {
+    const getCategoriesForTeam = makeGetCategoriesForTeam();
+
+    return memoizeResult((state: GlobalState) => {
+        const currentTeamId = getCurrentTeamId(state);
+        const categories = getCategoriesForTeam(state, currentTeamId);
+        return categories.filter((category) =>
```

Comment:
Also, perhaps this should be wrapped in a set to make checking membership a bit quicker?